app/models/maintenance_tasks/csv_collection_builder.rb

```diff
@@ -16,13 +16,13 @@ def collection(task)
     end
 
     # The number of rows to be processed. Excludes the header row from the
-    # count and assumes a trailing newline is at the end of the CSV file.
-    # Note that this number is an approximation based on the number of
-    # newlines.
+    # count and use the CSV library for an accurate row count.
+    # Note that the entire file is loaded. Files with millions of rows, it will
+    # take several seconds.
```

Comment:
```suggestion
    # count and uses the CSV library for an accurate row count.
    # Note that the entire file is loaded. It will take several seconds with files with millions of rows.
```